### ISSUE TITLE
[interactive-supports-focus] allow specifying directives in allowList

### ISSUE THREAD

**Issue by forivall** [angular-eslint/angular-eslint#1996]
**Description and reproduction of the issue**

Some directives provide the accessibility features that are otherwise reported by `interactive-supports-focus` and `click-events-have-key-events`. It should be possible to allowlist elements with these directives.

```JSON
{
  "rules": {
    "@angular-eslint/template/click-events-have-key-events": [
      "error",
      {
        "allowList": [
          "[mat-tab-link]"
        ]
      }
    ],
    "@angular-eslint/template/interactive-supports-focus": [
      "error",
      {
        "allowList": [
          "[mat-tab-link]"
        ]
      }
    ]
  }
}
```

(based on ["Tabs and navigation"](https://material.angular.io/components/tabs/overview#tabs-and-navigation)  example)

```HTML
<nav mat-tab-nav-bar [backgroundColor]="background">
  <a
    mat-tab-link
    *ngFor="let link of links"
    (click)="activeLink = link"
    [active]="activeLink == link"
  >
    {{link}}
  </a>
  <a mat-tab-link disabled>Disabled Link</a>
</nav>
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `17.5.2` |
| `@angular-eslint/template-parser`         | `17.5.2` |
| `@typescript-eslint/parser`               | `5.62.0` |
| `ESLint`                                  | `8.57.0` |
| `node`                                    | `20.13.1` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 17.3.8
Node: 20.13.1
Package Manager: npm 10.5.2
OS: darwin arm64

Angular: 17.3.11
... animations, common, compiler, compiler-cli, core, forms
... language-service, localize, platform-browser
... platform-browser-dynamic, platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1703.8
@angular-devkit/build-angular   17.3.8
@angular-devkit/core            17.3.8
@angular-devkit/schematics      17.3.8
@angular/cdk                    17.3.10
@angular/cli                    17.3.8
@angular/material               17.3.10
@schematics/angular             16.2.10
rxjs                            7.8.1
typescript                      5.3.3
zone.js                         0.13.3

```

This is a feature request for something that is not supported by the latest version, and I cannot upgrade versions in the project, so this is not really applicable

- [ ] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by jelledijkstra97** [comment]
+1

The '@angular-eslint/template/interactive-supports-focus' rule supports the allowList option but '@angular-eslint/template/click-events-have-key-events' does not. 

Can the allowList option be added to the click-events rule?

**Comment by JamesHenry** [comment]
Thanks I'd happily accept a PR for this.

It's a very similar discussion as here: https://github.com/angular-eslint/angular-eslint/issues/2193#issuecomment-2708479869

...in terms of what to call things. `allowList` seems a little too vague to me I think...

Please join in the discussion as I think it could form a mini standard across all these rules

**Comment by lucasousi** [comment]
@JamesHenry  The same issue occurs for `<div role="button" (click)="doSomething())></div>`. The lint warning appears (`click-events-have-key-events`), but if you put `(keydown.space)="doSomething()"` and `(keydown.enter)="doSomething()"`, the event will be executed twice because `role="button"` already handles those cases.

**Comment by JamesHenry** [comment]
@lucasousi Why would you be using `<div role="button"></div>` and not `<button></button>`?

**Comment by JamesHenry** [comment]
Having checked the rule and some of the other cases, I do agree that this one should not error so I'll address it, but... ewww :D

**Comment by JamesHenry** [comment]
https://github.com/angular-eslint/angular-eslint/pull/2488

**Comment by lucasousi** [comment]
> Having checked the rule and some of the other cases, I do agree that this one should not error so I'll address it, but... ewww :D

🤣🤣

I completely agree that using a `<button>` is the ideal approach. However, the world isn’t perfect 😅 — many projects out there have inconsistent code, and it’s not uncommon to see a `<div>` with `role="button"`. Even worse, some use plain `<div>s` with a `(click)` handler and no role at all (for this case, the warning is expected)

As much as it pains me to say it, for those that at least have the correct `role`, I think we have to accept them — even if it feels wrong.

It just feels strange to see a warning for something that actually works and is semantically acceptable.
